src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -462,6 +462,23 @@ class HttpGoogleServicesDAO(
     result
   }
 
+  def diagnosticBucketRead(user: RawlsUser, bucketName: String): Future[Option[ErrorReport]] = {
+    getUserCredentials(user) map { credentialOpt =>
+      credentialOpt match {
+        case None => Some(ErrorReport(StatusCodes.InternalServerError, "Unable to load credentials for user"))
+        case Some(credential) => {
+          val getter = getStorage(credential).objects().list(bucketName)
```

Comment:
note to self: list is most likely not the best thing to do to test if a user can read a bucket (particularly if there are a lot of files in the bucket). if we're calling this endpoint every time a user loads a workspace, we want it to be as quick as possible